products/messaging/backend/api/message_categories.py

```diff
@@ -0,0 +1,61 @@
+from rest_framework import serializers, viewsets
+from rest_framework.permissions import IsAuthenticated
+from posthog.api.forbid_destroy_model import ForbidDestroyModel
+from posthog.api.routing import TeamAndOrgViewSetMixin
+from posthog.models import MessageCategory
+
+
+class MessageCategorySerializer(serializers.ModelSerializer):
+    def validate(self, data):
+        # Ensure key is unique per team
+        if MessageCategory.objects.filter(team_id=self.context["team_id"], key=data["key"], deleted=False).exists():
+            raise serializers.ValidationError({"key": "A message category with this key already exists."})
+        return data
+
+    class Meta:
+        model = MessageCategory
+        fields = (
+            "id",
+            "key",
+            "name",
+            "description",
+            "public_description",
+            "category_type",
+            "created_at",
+            "updated_at",
+            "created_by",
+            "deleted",
+        )
+        read_only_fields = (
+            "id",
+            "created_at",
+            "updated_at",
+            "created_by",
+        )
+
+    def create(self, validated_data):
+        validated_data["team_id"] = self.context["team_id"]
+        validated_data["created_by"] = self.context["request"].user
+        return super().create(validated_data)
+
+
+class MessageCategoryViewSet(
+    TeamAndOrgViewSetMixin,
+    ForbidDestroyModel,
+    viewsets.ModelViewSet,
+):
+    scope_object = "INTERNAL"
+    permission_classes = [IsAuthenticated]
```

Comment:
Feels like this isn't necessary right? This is private so the TeamAndOrgViewSetMixin covers all the auth necessary